### PR TITLE
fixes #9023 and #9025 - Re-adds passdoor flag interaction that alexkar removed because he can't be bothered to actually read

### LINKED ISSUE #9023: Drone's problem
Issue reported from  Round ID: 30233 (Yogstation 13)

Reporting client version: 513.1525

hello, I had a problem today with the drone, because it can't get under the airlocks, so opening them up instead of going underneath, I don't know if this is a bug but it gets in the way of the gameplay as a drone. I hope you fix this, as the gameplay with the Drone is supposed to be something "discreet", but with it opening airlocks can make it difficult. I appreciate the possible help :)

### LINKED ISSUE #9025: Rats cant walk under doors
Title

### PR DESCRIPTION
**Pull request by jcatjmeow**
### Intent of your Pull Request
fixes #9023 
fixes #9025

As title.
#### Changelog

:cl:  
bugfix: passdoor now works again. Drones, mice etc can fit under doors again.
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -132,8 +132,12 @@
...
 /obj/machinery/door/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
-	if(istype(mover) && (mover.pass_flags & PASSGLASS))
-		return !opacity
+	if(istype(mover)) //yogs start
+		if(mover.pass_flags & PASSGLASS)
+			return !opacity
+		else if(mover.pass_flags & PASSDOOR)
+			return TRUE //yogs end
+	return .
```
Comment (Bibby0110) [inline code/game/machinery/doors/door.dm:140]: redundant

Comment (jcatjmeow) [reply code/game/machinery/doors/door.dm:140]: ?

Comment (jcatjmeow) [reply code/game/machinery/doors/door.dm:140]: maybe I dunno

**Comment by Takahiru** [comment]
hc cant read best pr

File: code/game/machinery/doors/door.dm (hunk 1 above)
Comment (Bibby0110) [reply code/game/machinery/doors/door.dm:140]: Redundant as in remove it

A proc automatically returns ., no need to do it explicitly

**Review by fluffe9911 — APPROVED** [review]
ha ha alexkar cant read

Edit: also I saw gamers thing but I dont care

**Pull request merged**